components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/design-view/design-grid.js

```diff
@@ -656,10 +656,23 @@ define(['require', 'log', 'jquery', 'backbone', 'lodash', 'designViewUtils', 'dr
                     * explicitly set the sourceId.  Simply if sourceId is '' that means this connection is related to a
                     * partition.
                     * */
+                    var isFromFaultStream = connection.sourceId.substr(connection.sourceId.indexOf('-')) === '-err-out';
                     if (sourceId === '') {
                         sourceId = source;
                         sourceType = 'PARTITION';
                     } else {
+                        if (isFromFaultStream) {
+                            // Change the source id to the respective source id of the fault stream
+                            var streamMap = {};
+                            var eventStreamName = '';
+                            self.configurationData.getSiddhiAppConfig().getStreamList().forEach(function(s) {
+                                streamMap[s.getName()] = s;
+                                if (s.getId() === sourceId) {
+                                    eventStreamName = s.getName();
+                                }
+                            });
+                            sourceId = streamMap['!' + eventStreamName].getId();
```

Comment:
shall we use the defined constant for this

Reply:
That constant has been added at the Java level. Will introduce a new one in JS level and use that here.

Reply:
Fixed in commit https://github.com/wso2/carbon-analytics/pull/1587/commits/7875d09ce7453c9f38bf7c70afe12336016d8908